cg/cli/store/fastq.py

```diff
@@ -1,84 +1,131 @@
+        context=context, dry_run=dry_run, sample_ids=[sample.internal_id for sample in samples]
+    )
     LOG.info(f"Stored fastq files for {stored_individuals} samples")
 
 
 @click.command("ticket")
 @click.argument("ticket", type=str)
-@click.option("-d", "--dry-run", is_flag=True)
+@DRY_RUN
 @click.pass_context
-def store_ticket(context: click.Context, ticket: str, dry_run: bool):
-    """Include links to decompressed FASTQ files belonging to this ticket in Housekeeper"""
+def store_ticket(context: click.Context, ticket: str, dry_run: bool) -> None:
+    """Include links to decompressed FASTQ files belonging to a ticket in Housekeeper."""
     status_db: Store = context.obj.status_db
-    samples: List[models.Sample] = status_db.get_samples_from_ticket(ticket=ticket)
-    stored_individuals = 0
+    samples: List[Sample] = status_db.get_samples_from_ticket(ticket=ticket)
+    stored_individuals: int = invoke_store_samples(
+        context=context, dry_run=dry_run, sample_ids=[sample.internal_id for sample in samples]
+    )
+    LOG.info(f"Stored fastq files for {stored_individuals} samples")
+
+
+@click.command("bundles")
+@click.argument("flow-cell-id", type=str)
+@DRY_RUN
+@click.pass_obj
+def store_bundles(context: click.Context, flow_cell_id: str, dry_run: bool) -> None:
+    """Include all files in a flow cell bundle and its corresponding sequencing files. Updates SPRING metadata file"""
+    compress_api: CompressAPI = context.meta_apis["compress_api"]
+    status_db: Store = context.status_db
+    update_compress_api(compress_api, dry_run=dry_run)
+
+    samples: List[Sample] = status_db.get_samples_from_flow_cell(flow_cell_id=flow_cell_id)
+    bundle_names: List[str] = (
+        [sample.internal_id for sample in samples if sample.internal_id] + [flow_cell_id]
+        if samples
+        else [flow_cell_id]
+    )
+    for bundle_name in bundle_names:
+        compress_api.hk_api.include_files_to_latest_version(bundle_name=bundle_name)
+
+    if not samples:
+        LOG.info(f"No samples found for flow cell {flow_cell_id} in the database")
+        return
     for sample in samples:
-        stored_count: int = context.invoke(
-            store_sample, sample_id=sample.internal_id, dry_run=dry_run
+        spring_metadata_file: File = compress_api.hk_api.get_file_from_latest_version(
+            bundle_name=sample.internal_id, tags=[SequencingFileTag.SPRING_METADATA]
         )
-        stored_individuals += stored_count
-    LOG.info(f"Stored fastq files for {stored_individuals} samples")
+        if spring_metadata_file:
+            LOG.info(
+                f"Updating file paths in SPRING metadata file {spring_metadata_file.full_path}: for sample: {sample.internal_id}"
+            )
+            update_metadata_paths(
+                spring_metadata_path=spring_metadata_file.full_path,
+                new_parent_path=Path(spring_metadata_file.full_path).parent,
+            )
```

Comment:
Do we not have multiple spring_metadata files per sample? I feel like an extra loop is needed here.

Reply:
Good catch! Changed to loop over multiple files